lua/core/menu/select.lua

```diff
@@ -1,37 +1,64 @@
+tabutil = require "tabutil"
+
 local m = {
   pos = 0,
   list = {},
-  len = "scan" 
+  favorites = {},
+  len = "scan"
 }
 
+local function menu_table_entry(file)
+  local p = string.match(file,".*/")
+  local n = string.gsub(file,'.lua','/')
+  n = string.gsub(n,_path.code,'')
+  n = string.sub(n,0,-2)
+  local a,b = string.match(n,"(.+)/(.+)$") -- strip similar dir/script
+  if a==b and a then n = a end
+  return {name=n,file=file,path=p}
+end
+
 local function sort_select_tree(results)
+  if tab.count(m.favorites) > 0 then
+    for _, entry in pairs(m.favorites) do
+      table.insert(m.list,entry)
+    end
+    table.insert(m.list, {name="-", file=nil, path=nil})
+  end
+
   local t = {}
   for filename in results:gmatch("[^\r\n]+") do
-    if string.match(filename,"/data/")==nil and 
+    if string.match(filename,"/data/")==nil and
       string.match(filename,"/lib/")==nil then
       table.insert(t,filename)
     end
   end
 
   for _,file in pairs(t) do
-    local p = string.match(file,".*/")
-    local n = string.gsub(file,'.lua','/')
-    n = string.gsub(n,_path.code,'')
-    n = string.sub(n,0,-2)
-    local a,b = string.match(n,"(.+)/(.+)$") -- strip similar dir/script
-    if a==b and a then n = a end
-    --print(file,n,p)
-    table.insert(m.list,{name=n,file=file,path=p})
+    table.insert(m.list,menu_table_entry(file))
   end
 
   m.len = tab.count(m.list)
   _menu.redraw()
 end
 
+local function contains(list, menu_item)
+  for _, v in pairs(list) do
+    if v.file == menu_item.file then
+      return true
+    end
+  end
+  return false
+end
 
 m.init = function()
-  m.len = "scan" 
+  m.len = "scan"
   m.list = {}
+  m.favorites = {}
+  m.favorites = tabutil.load("/home/we/dust/.favorites")
```

Comment:
UPDATE: after some thought is would be valuable to call this file `system.favorites` so that it is visible in maiden and sits nicely along side the existing `system.state` and `system.pset` files

Reply:
agreed with @ngwese it'd be great to have this file be visible

otherwise, this is great!

Reply:
That all makes sense to me, I'll try and get a patch submitted this evening or tomorrow morning.